posthog/tasks/subscriptions.py

```diff
@@ -55,21 +72,88 @@ def send_email_subscription_report(
         subject=subject,
         template_name="subscription_report",
         template_context={
-            "images": [exported_asset.get_public_content_url()],
+            "images": [x.get_public_content_url() for x in assets],
             "resource_noun": resource_noun,
             "resource_name": resource_name,
-            "resource_url": resource_url,
-            "subscription_url": subscription.url,
+            "resource_url": f"{resource_url}?{UTM_TAGS}",
+            "subscription_url": f"{subscription.url}?{UTM_TAGS}",
             "unsubscribe_url": get_unsubscribe_url(subscription=subscription, email=email),
             "inviter": inviter if is_invite else None,
             "self_invite": self_invite,
             "invite_message": invite_message,
+            "total_asset_count": total_asset_count,
         },
     )
     message.add_recipient(email=email)
     message.send()
 
 
+def _deliver_subscription_report(
+    subscription_id: int, new_emails: Optional[List[str]] = None, invite_message: Optional[str] = None
+) -> None:
+    is_invite = new_emails is not None
+
+    if is_invite and not new_emails:
+        return
+
+    subscription = (
+        Subscription.objects.prefetch_related("dashboard__insights")
+        .select_related("created_by", "insight", "dashboard",)
+        .get(pk=subscription_id)
+    )
+
+    if subscription.target_type == "email":
+        insights = []
+
+        if subscription.dashboard:
+            tiles = get_tiles_ordered_by_position(subscription.dashboard)
+            insights = [tile.insight for tile in tiles]
+        elif subscription.insight:
+            insights = [subscription.insight]
+        else:
+            raise Exception("There are no insights to be sent for this Subscription")
+
+        # Create all the assets we need
+        assets = [
+            ExportedAsset(team=subscription.team, export_format="image/png", insight=insight)
+            for insight in insights[:MAX_ASSET_COUNT]
+        ]
+        ExportedAsset.objects.bulk_create(assets)
+
+        # Wait for all assets to be exported
+        tasks = [export_task.s(asset.id) for asset in assets]
+        parallel_job = group(tasks).apply_async()
+
+        max_wait = 30
+        while not parallel_job.ready():
+            max_wait = max_wait - 1
+            sleep(1)
+            if max_wait < 0:
+                raise Exception("Timed out waiting for exports")
+
+        # Send emails
+        emails_to_send = new_emails or subscription.target_value.split(",")
+
+        for email in emails_to_send:
+            try:
+                send_email_subscription_report(
+                    email,
+                    subscription,
+                    assets,
+                    invite_message=invite_message or "" if is_invite else None,
+                    total_asset_count=len(insights),
+                )
+            except Exception as e:
+                logger.error(e)
```

Comment:
Could a statsd signal help here so we could graph and alert on failing subscriptions?